### PR TITLE
Update json schema

### PR DESCRIPTION
**Pull request by smohiudd**
Update form json schema to include:
- providers
- renders
- item-assets
- assets (thumbnail)

### CODE REVIEW

File: FormSchemas/jsonschema.json [hunk 1]
```diff
@@ -122,6 +183,130 @@
...
+        ]
+      }
+    },
+    "required": [
+      "cog_default"
+      ]
+    },
+    "renders": {
+      "type": "object",
+      "title": "Renders"
```
Comment (stephenkilbourn) [inline FormSchemas/jsonschema.json:233]: looks like this is missing a type

File: FormSchemas/jsonschema.json [hunk 2]
```diff
@@ -122,6 +183,130 @@
...
+            "type": "string",
+            "format": "iri"
+          }
+        }
+      }
+    },
+    "assets": {
+      "type": "object",
+      "properties": {
+        "thumbnail": {
```
Comment (stephenkilbourn) [inline FormSchemas/jsonschema.json:277]: is thumbnail really the only property for the assets object?

**Comment by stephenkilbourn** [comment]
Just a few questions to make sure the fields are correct.

I added a commit that styles the form to make the new fields look a little better.

**Review by stephenkilbourn — DISMISSED** [review]

File: FormSchemas/jsonschema.json (hunk 2 above)
Comment (smohiudd) [reply FormSchemas/jsonschema.json:277]: There could be others, but thumbnails is the only we use in the catalog right now.

File: FormSchemas/jsonschema.json [hunk 3]
```diff
@@ -91,13 +93,72 @@
...
+          "id_regex": {
+            "type": "string",
+            "title": "Id Regex"
+          },
+          "id_template": {
+            "type": "string",
+            "title": "Id Template"
+          },
+          "use_multithreading":{
+            "type": "string",
```
[the remark this answers is not in the document]
Comment (smohiudd) [reply FormSchemas/jsonschema.json:159]: yeah you're right should be boolean.

File: FormSchemas/jsonschema.json (hunk 1 above)
Comment (smohiudd) [reply FormSchemas/jsonschema.json:233]: The renders object follows this schema: https://github.com/stac-extensions/render/blob/main/json-schema/schema.json

Do you think we should use that schema or just make it an object where someone can copy paste their renders object?

Comment (stephenkilbourn) [reply FormSchemas/jsonschema.json:233]: without any properties defined, we will just get an empty fieldset from React-json-shema
Screenshot 2024-12-19 at 5 37 08 PM

I think the library will support references within the json schema, so I can try copying the file to this repo, adding the reference, and styling it like the others.

Comment (stephenkilbourn) [reply FormSchemas/jsonschema.json]: @smohiudd - It turns out that the react-jsonschema-form library "only supports local definition referencing". There is a separate [json-schema-ref-parser](https://github.com/APIDevTools/json-schema-ref-parser) library that does let us parse the json schema. I pulled that in and put a copy of the stac-extensions schema above in the public directory to try it out.  Unfortunately, it seems to be hanging up with some references.  I'm still trying to debug, but I pushed a commit with a partially working version.

Another work around would be something like allow some or all of the Renders object be a text area where users paste in the object.

Comment (stephenkilbourn) [reply FormSchemas/jsonschema.json]: ok, I made that change and pushed it up.  I'm good with merging and trying it this way.

Comment (smohiudd) [reply FormSchemas/jsonschema.json]: Thanks!

**Pull request merged**